### PR TITLE
Jet dqm test 740 pre1

### PR DESCRIPTION
**Pull request by rkunnawa**

### CODE REVIEW

**Comment by rkunnawa** [comment]
This is commit of the HeavyIon jet DQM/Validation suites. this module reads over HI background subtracted jets and creates plots showing any issue with the any step of the subtraction method by reading the candidate information for both PF and Calo jets.

**Comment by davidlange6** [comment]
please test

**Comment by degano** [comment]
@cmsbuild please test

**Comment by rkunnawa** [comment]
The issue should be fixed now with the buildfile updated with the location of the CentralityProvider header file. (although i didnt see this issue when i tried a fresh compilation test this morning with 740_pre5)

**Comment by degano** [comment]
@cmsbuild please test

**Comment by rkunnawa** [comment]
The main issue was related to the centrality provider header file which became obsolete as of a few days. so the issue is fixed and i checked it in the same IB and it compiled fine. hope this can get tested again. thanks

**Comment by degano** [comment]
@cmsbuild please test

**Comment by deguio** [comment]
@rkunnawa could you summarize the content of this PR?
thanks,
F.

**Comment by deguio** [comment]
btw, it seems that the PR is not mergeable anymore. a rebase is probably needed.
F.

**Comment by rkunnawa** [comment]
Hi @deguio, this PR contains the Heavy Ion jet DQM/Validation modules. they specifically look at the background subtracted jets and point out if there is anything going wrong in (the many steps) of the subtraction. It is very different from the pp modules already present which looks at PF or Calo jets alone, hence necessitating a separate module.

**Comment by slava77** [comment]
needs a rebase?

**Comment by yetkinyilmaz** [comment]
This pull-request is rebased now, can somebody have a look if it's ok now?
@slava77 @deguio

**Comment by slava77** [comment]
did you push after the rebase?
this one seems the same as an hour ago.

**Comment by yetkinyilmaz** [comment]
I did, to me it appears indeed different now, and the last commit is 
1b9653c1636feb342ef2230177067f92e8cc9477

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/7405	1b9653c
changes are irrelevant to reco, the signature is needed because of a bad rebase.

@yetkinyilmaz 
I was added into this PR only after 1b9653c appeared and it doesn't look like the rebase went well: it's showing all merges from the start of the original topic branch.

After merging locally, manually, I see the following changes on top of CMSSW_7_4_X:

```
 DQMOffline/JetMET/BuildFile.xml                           |    7 +
 DQMOffline/JetMET/interface/JetAnalyzer_HeavyIons.h       |  336 ++++++++++++++++++++++++++
 DQMOffline/JetMET/plugins/BuildFile.xml                   |   13 +
 DQMOffline/JetMET/python/jetAnalyzer_cff.py               |    9 +-
 DQMOffline/JetMET/python/jetAnalyzer_cfi.py               |  277 +++++++++++++++++++++
 DQMOffline/JetMET/src/JetAnalyzer_HeavyIons.cc            | 1136 ++++++++++++++++++++++
 DQMOffline/JetMET/src/SealModule.cc                       |    2 +
 Validation/RecoHI/python/globalValidationHeavyIons_cff.py |    2 +-
 Validation/RecoJets/BuildFile.xml                         |    2 +
 Validation/RecoJets/plugins/BuildFile.xml                 |    3 +
 Validation/RecoJets/plugins/JetTester_HeavyIons.cc        | 1190 +++++++++++++++++++++++++++++++
 Validation/RecoJets/plugins/JetTester_HeavyIons.h         |  336 ++++++++++++++++++++++++++
 Validation/RecoJets/plugins/SealModule.cc                 |    3 +
 Validation/RecoJets/python/JetValidationHeavyIons_cff.py  |  524 +++++++++++++++++
 14 files changed, 3837 insertions(+), 3 deletions(-)


```

**Comment by deguio** [comment]
ciao @rkunnawa 
I am reviewing your PR. my first quick comments are:
- there are TH1* instantiated with new which are not deleted. please fix the memory leaks by adding the corresponding delete.
- the beginJob and endJob methods are not called by the framework. please just remove them
- it seems to me that the two classes (JetTester and JetAnalyzer) are very similar. is there a reason not to make a unique class to be used in both the DQM and the Valid sequences since the gen quantities are not used?

thanks,
F.

**Comment by rkunnawa** [comment]
hi @deguio , thanks for your comments. 
I will update the code to fix the TH1's and the begin/end Job methods. 
Regarding the two classes, the JetTester_HeavyIons and JetAnalyzer_HeavyIons in the validation and DQMOffline modules are necessary because we are going to add in gen related variables soon to the validation one (as an addition to the next iteration with the setup as it is). 

a possible question: 
Since i made this branch in an "old" setup and any commits would require another rebase yeah? I made a new branch of the same HeavyIon DQM/Validation modules on the latest IB. I can initiate a PR for that new branch (along with the changes you suggested) then if its easier to compare and check. but if its not possible for a new PR to get into the next release then we can go with this.

**Comment by ianna** [comment]
@rkunnawa - please, rebase the PR.

**Comment by rkunnawa** [comment]
superseded by PR 7527 on a clean branch (to remove all the issues about), thanks
